developer/src/kmc-kmn/src/kmw-compiler/validate-layout-file.ts

```diff
@@ -144,9 +144,11 @@ function CheckKey(
     // Keyman versions before 14 do not support '*special*' labels on non-special keys.
     // ZWNJ use, however, is safe because it will be transformed in function
     // TransformSpecialKeys14 to '<|>',  which does not require the custom OSK font.
-    if((CSpecialText10.includes(FText) || CSpecialText14.includes(FText)) &&
+    if((CSpecialText10.includes(FText) || CSpecialText14.includes(FText) || CSpecialText17.includes(FText)) &&
         !CSpecialText14ZWNJ.includes(FText) &&
         !IsKeyboardVersion14OrLater() &&
+        !CSpecialText17ZWNJ.includes(FText) &&
+        !IsKeyboardVersion17OrLater() &&
```

Comment:
We need to split this test because it's not possible to test all in one go like this. It may be best to build a combined array of all possible values based on the version, so V17 arrays would include V14 (but not vice-versa).

Reply:
ok, I made an attempt with your suggested map...